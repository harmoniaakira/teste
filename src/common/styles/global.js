import { createGlobalStyle } from 'styled-components'

const globalStyle = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  body {
    font-family: 'Red Hat Display', -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;
    margin: 0;
  }
`

export default globalStyle
