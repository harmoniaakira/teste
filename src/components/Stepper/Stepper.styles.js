import styled from 'styled-components'
import NavigateBeforeIcon from '@material-ui/icons/NavigateBefore';
import InfoOutlinedIcon from '@material-ui/icons/InfoOutlined';

export const MobileStepperContainer = styled.div`
  display: block;
  width: 100%;
  max-width: 768px;

  @media (min-width: 1440px) {
    display: none;
  }
`

export const WebStepperContainer = styled.div`
  display: none;
  position: absolute;
  top: 436px;
  left: 90px;

  @media (min-width: 1440px) {
    display: block;
  }
`

export const StepperHeader= styled.div`
  // display: flex;
  // justify-content: space-between;
  // padding: 0 14px;

  height: 56px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0px 20px;
  position: relative;
  background: #F0F0F0;
`

export const StepperTitle= styled.p`
  color: #8E8E93;
  font-weight: 700;
  font-size: 12px;
  line-height: 16px;
  letter-spacing: 0.5px;
  text-transform: uppercase;
`

export const StepperText = styled.p`
  color: #8E8E93;
  font-size: 12px;
  line-height: 14px;
  font-weight: 400;

  @media (min-width: 1440px) {
  color: #FFFFFF;
    font-size: 18px;
    line-height: 28px;
    font-weight: 400;
  }
`

export const Steps = styled.div`
  display: flex;
`

export const Step = styled.div`
  background: ${
      props => props.done
      ? "rgb(140, 209, 51);" 
      : props.active 
        ? "linear-gradient(90deg, rgb(195, 195, 195) 0%, rgb(195, 195, 195) 50%, rgb(224, 224, 224) 50%);" 
        : "rgb(224, 224, 224);"
    };
    width: 25%;
    height: 4px;
    margin-top: 10px;
    &:not(:last-child) {
      margin-right: 6px;
    }

  @media (min-width: 1440px) {
    background: ${
      props => props.done
      ? "rgba(255, 255, 255, 0.7);" 
      : props.active 
        ? "linear-gradient(90deg, rgba(255,255,255,0.7) 0%, rgba(255,255,255,0.7) 50%, rgba(255,255,255,0.3) 50%);" 
        : "rgba(255, 255, 255, 0.3);"
    };
    width: 70px;
    height: 5px;
    margin-right: 5px;
    margin-top: 15px;
  }
`

export const BackButton = styled.img`
`

export const InfoButton = styled.img`
`

export const MobileHeader = styled.div`
    height: 56px;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 0px 20px;
    position: relative;
    // background: #F0F0F0;

`

export const MobileHeaderText = styled.div`
    display: flex;
    flex-direction: column;
    text-align: center;
    
    color: #8E8E93;
    font-size: 12px;
`

export const MobileHeaderTextTitle = styled.div`
    font-weight: bold;
    letter-spacing: 0.5px;
    text-transform: uppercase;
    line-height: 15.88px;
`

export const MobileHeaderTextStep = styled.div`
    line-height: 13px;
`

export const MobileHeaderStepper = styled.div`
    position: absolute;
    width: 100%;
    bottom: 0;
    left: 0;
`

export const ArrowBack = styled(NavigateBeforeIcon)`
    height: 40px;
    width: 40px;
    color: #054375;
    cursor: pointer;
`

export const Info = styled(InfoOutlinedIcon)`
    width: 20px;
    height: 20px;
    color: #054375;
    cursor: pointer;
`