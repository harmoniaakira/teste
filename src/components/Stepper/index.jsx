import React, { memo, useEffect, useState } from 'react';
import { useStepperContext } from 'contexts/StepperContext';
import PropTypes from 'prop-types';
import { useHistory } from "react-router-dom";
import {
    ArrowBack, Info, MobileHeader, MobileHeaderStepper, MobileHeaderText, MobileHeaderTextStep, MobileHeaderTextTitle, MobileStepperContainer, Step, StepperText,
    Steps, WebStepperContainer
} from './Stepper.styles';


const Stepper = ({ infoCallback, arrowCallback }) => {
    const history = useHistory();
    const { totalSteps, currentStep, setCurrentStep } = useStepperContext()
    const [stepsArray, setStepsArray] = useState([]);

    const back = () => {
        if (typeof arrowCallback === "function") {
            arrowCallback()
        } else {
            history.goBack();
        }
    }

    const renderInfo = () => {
        if (typeof infoCallback === "function") {
            infoCallback()
        }
    }

    const renderTitleByStep = () => {
        switch (currentStep) {
            default:
            case 1:
                return 'Informações do Cliente'
            case 2:
                return 'Código Validação'
            case 3:
                return 'Prova de Vida'
            case 4:
                return 'Captura de Documento'
        }
    }

    const RenderSteps = () => (
        <Steps>
            {
                stepsArray.map((step) => {
                    return step;
                })
            }
        </Steps>
    )
    
    const handleUpdateCurrStep = (pathname) => {
        switch (pathname) {
            case "/onboarding-pf-frontend/":
            case "/onboarding-pf-frontend/cadastro":
            default:
                setCurrentStep(1)
                break;
            case "/onboarding-pf-frontend/envio-de-codigo":
                setCurrentStep(2)
                break;
            case "/onboarding-pf-frontend/foto":
                setCurrentStep(3)
                break;

            case "/onboarding-pf-frontend/envio-de-documento":
                setCurrentStep(4)
                break;
        }
    }

    useEffect(() => {
        const aux = [];
        for (let i = 1; i <= totalSteps; i++) {
            if (i === currentStep) {
                aux.push(<Step key={i} active />)
            } else if (i < currentStep) {
                aux.push(<Step key={i} done />)
            } else {
                aux.push(<Step key={i} />)
            }
        }
        setStepsArray(aux);
    }, [totalSteps, currentStep])

    useEffect(() => {
        const { location } = history
        handleUpdateCurrStep(location.pathname)
    }, [history])

    return <>
        <MobileStepperContainer>
            <MobileHeader>
                <ArrowBack onClick={back} />
                <MobileHeaderText>
                    <MobileHeaderTextTitle>{renderTitleByStep()}</MobileHeaderTextTitle>
                    <MobileHeaderTextStep> Etapa {currentStep} de {totalSteps} </MobileHeaderTextStep>
                </MobileHeaderText>
                <Info onClick={renderInfo} disabled={typeof infoCallback !== "function"} />
                <MobileHeaderStepper>
                    <RenderSteps />
                </MobileHeaderStepper>
            </MobileHeader>
        </MobileStepperContainer >
        <WebStepperContainer>
            <StepperText><strong>Etapa {currentStep} de {totalSteps}:</strong> Dados do cliente</StepperText>
            <RenderSteps />
        </WebStepperContainer>
    </>
}

Stepper.propTypes = {
    infoCallback: PropTypes.func,
    arrowCallback: PropTypes.func,

}

Stepper.defaultProps = {
    infoCallback: undefined,
    arrowCallback: undefined
}

export default memo(Stepper)