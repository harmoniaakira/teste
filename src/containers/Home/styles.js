import styled from 'styled-components'
import { Typography } from '@material-ui/core'

export const Container = styled.div`
  background: ${({ theme }) => theme.palette.primary.background};
  height: 100vh;
  text-align: center;
`

export const Title = styled(Typography)`
  margin: 0;
  margin-bottom: 10px;
`
