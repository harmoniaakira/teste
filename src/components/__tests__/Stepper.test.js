import React from 'react';
import Stepper from 'components/Stepper';
import { ArrowBack, Info, MobileHeaderStepper } from 'components/Stepper/Stepper.styles';
import { render, shallow } from 'enzyme';


const mockHistoryBack = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        goBack: mockHistoryBack,
      }),
}));

describe("Stepper", () => {
    const stepper = shallow(<Stepper />)

    it('should render step', () => {
        expect(stepper.find(MobileHeaderStepper).children()).toHaveLength(1)
    });
    
    it('should call history goback', () => {
        stepper.find(ArrowBack).props().onClick();
        expect(mockHistoryBack.mock.calls.length).toEqual(1);
    })

    it('should match snapshot', () => {
        stepper.find(Info).props().onClick();
        render(<Stepper />)
        expect(stepper).toMatchSnapshot()
    })
    
})