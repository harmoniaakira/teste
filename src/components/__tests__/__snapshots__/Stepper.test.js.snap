// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper should match snapshot 1`] = `
@media (min-width:1440px) {

}

@media (min-width:1440px) {

}

@media (min-width:1440px) {

}

<Fragment>
  <styled.div>
    <styled.div>
      <Styled(Component)
        onClick={[Function]}
      />
      <styled.div>
        <styled.div>
           Informações do Cliente 
        </styled.div>
        <styled.div>
           Etapa 
          1
           de 
          4
           
        </styled.div>
      </styled.div>
      <Styled(Component)
        disabled={true}
        onClick={[Function]}
      />
      <styled.div>
        <RenderSteps />
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.p>
      <strong>
        Etapa 
        1
         de 
        4
        :
      </strong>
       Dados do cliente
    </styled.p>
    <RenderSteps />
  </styled.div>
</Fragment>
`;
